src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewDataInfo.cs

```diff
@@ -10,6 +10,7 @@ public class ViewDataInfo
     {
         private object _value;
         private Func<object> _valueAccessor;
+        private Func<PropertyInfo, object, object> _stateValueAccessor;
```

Comment:
I like the general thought but am not sure `_stateValueAccessor` is needed.

What we need is primarily an indication that we haven't tried `PropertyInfo.GetValue(Container)` yet. Could use a unique (i.e. `static`) `Func<object>` that the new constructor, now `ViewDataInfo(object, PropertyInfo)`, uses. Or, don't bother w/ a new constructor and use `valueAccessor ?? IndicatorFunc` on [line 33](https://github.com/aspnet/Mvc/pull/5325/files#diff-04d8b5d1afe85772984a16dd42af2555R33) and pass `null` in from `ViewDataEvaluator`.

Either way, use `ReferenceEquals(IndicatorFunc, _valueAccessor)` in the existing `_valueAccessor != null` block then call `PropertyInfo.GetValue(Container)`.